test/Errors.test.js

```diff
@@ -1,363 +1,488 @@
-				resolve();
+function compile(options) {
+	return new Promise((resolve, reject) => {
+		const compiler = webpack(
+			Object.assign(
+				{
+					context,
+					devtool: false,
+					optimization: { minimizer: [] }
+				},
+				options
+			)
+		);
+		compiler.outputFileSystem = outputFileSystem;
+		compiler.run((err, stats) => {
+			if (err) return reject(err);
+			compiler.close(err2 => {
+				if (err2) return reject(err2);
+				const { errors, warnings } = stats.toJson({ errorDetails: false });
+				resolve({ errors, warnings });
 			});
 		});
-	}
-	it("should throw an error if file doesn't exist", done => {
-		getErrors(
-			{
-				mode: "development",
-				entry: "./missingFile"
-			},
-			(errors, warnings) => {
-				expect(errors).toHaveLength(2);
-				expect(warnings).toHaveLength(0);
-				let lines = errors[0].split("\n");
-				expect(lines[0]).toMatch(/missingFile.js 4:0/);
-				expect(lines[1]).toMatch(/^Module not found/);
-				expect(lines[1]).toMatch(/\.\/missing/);
-				lines = errors[1].split("\n");
-				expect(lines[0]).toMatch(/missingFile.js 12:9/);
-				expect(lines[1]).toMatch(/^Module not found/);
-				expect(lines[1]).toMatch(/\.\/dir\/missing2/);
-				done();
-			}
-		);
 	});
-	it("should report require.extensions as unsupported", done => {
-		getErrors(
-			{
+}
+
+it("should throw an error if file doesn't exist", async () => {
+	await expect(
+		compile({
+			mode: "development",
+			entry: "./missingFile"
+		})
+	).resolves.toMatchInlineSnapshot(`
+Object {
+  errors: Array [
+    ./missingFile.js 4:0-20
+Module not found: Error: Can't resolve './missing' in 'test/fixtures/errors',
+    ./missingFile.js 12:9-34
+Module not found: Error: Can't resolve './dir/missing2' in 'test/fixtures/errors',
+  ],
+  warnings: Array [],
+}
+`);
+});
+it("should report require.extensions as unsupported", async () => {
+	await expect(
+		compile({
+			mode: "development",
+			entry: "./require.extensions"
+		})
+	).resolves.toMatchInlineSnapshot(`
+Object {
+  errors: Array [],
+  warnings: Array [
+    ./require.extensions.js 1:0-18
+require.extensions is not supported by webpack. Use a loader instead.,
+  ],
+}
+`);
+});
+it("should report require.main.require as unsupported", async () => {
+	await expect(
+		compile({
+			mode: "development",
+			entry: "./require.main.require"
+		})
+	).resolves.toMatchInlineSnapshot(`
+Object {
+  errors: Array [],
+  warnings: Array [
+    ./require.main.require.js 1:0-20
+require.main.require is not supported by webpack.,
+  ],
+}
+`);
+});
+it("should report module.parent.require as unsupported", async () => {
+	await expect(
+		compile({
+			mode: "development",
+			entry: "./module.parent.require"
+		})
+	).resolves.toMatchInlineSnapshot(`
+Object {
+  errors: Array [],
+  warnings: Array [
+    ./module.parent.require.js 1:0-21
+module.parent.require is not supported by webpack.,
+  ],
+}
+`);
+});
+it("should warn about case-sensitive module names", async () => {
```

Comment:
This test case depends on the type of filesystem (case-sensitivity or not?).

Here is how you could test it: https://github.com/webpack/webpack/blob/master/test/cases/errors/case-sensitive/test.filter.js

Either skip the test on the wrong filesystem or offer two alternative snapshots depending on filesystem

Reply:
yes, I've since added this - via `fs.existsSync`